app/models/decision-activity.js

```diff
@@ -1,10 +1,20 @@
-import Model, { attr, hasMany } from '@ember-data/model';
+import { attr, hasMany } from '@ember-data/model';
+import AbstractValidationModel from './abstract-validation-model';
+import Joi from 'joi';
+import { validateHasManyOptional } from '../validators/schema';
 
-export default class DecisionActivityModel extends Model {
+export default class DecisionActivityModel extends AbstractValidationModel {
   @attr('date') endDate;
 
   @hasMany('decision', {
     inverse: 'hasDecisionActivity',
   })
   givesCauseTo;
+
+  get validationSchema() {
```

Comment:
You are correct that the actual validation is never used. In the [`saveDecision`](https://github.com/lblod/frontend-organization-portal/blob/d9896e025a9b1f06ac2f786a419b33b3876616e9/app/controllers/administrative-units/administrative-unit/change-events/new.js#L320) function the presence of an `endDate` is checked manually. I will have a closer look into this to check whether the validation is needed and to ensure the correct data elements are actually saved.

Reply:
@aliokan I ended up removing this unused validation, after verifying things work as expected with respect to saving decision activities. Arguably, it could be used to ensure that the processed user input is a valid date. But I have noticed that invalid/incomplete inputs for optional dates are silently ignored throughout the frontend, so imo it is not worth the additional code complexity at this moment.